cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/environment/MasterEnvironment.java

```diff
@@ -53,11 +54,11 @@ default void destroy() {
   }
 
   /**
-   * Returns a {@link Optional} {@link MasterEnvironmentTask} to be executed periodically.
-   * It is guaranteed that there is no concurrent call to the task returned.
+   * Returns a list of {@link MasterEnvironmentTask}s to be executed periodically.
+   * It is guaranteed that there is no concurrent call to each task returned.
```

Comment:
I don't think the current implementation is not guaranteeing no concurrent task execution?

Reply:
I am not too sure myself. The underlying AbstractSchedulerService provides a guarantee that the run iterations of a specific task will never run concurrently. However we can create multiple instances of the same task and schedule them all to run concurrently.

Reply:
Let me just remove the guarantee claim.